plugins/workspace-minimap/src/minimap.ts

```diff
@@ -183,4 +190,18 @@ export class Minimap {
     private onMouseMove(event: PointerEvent): void {
       this.primaryScroll(event);
     }
+
+    /**
+     * Turns on focus mode; A highlight on the user's viewport in the minimap.
+     */
+    enableFocusMode(): void {
+      this.focusMode.init();
+    }
+
+    /**
+     * Turns off focus mode.
+     */
+    disableFocusMode(): void {
+      this.focusMode.dispose();
+    }
```

Comment:
Thinking about this from an API design perspective, I think taking in a boolean to enable/disable the focus mode might be a better option. If a developer is modifying the focus mode, they're probably doing that based on another boolean they already have.

For example:
```js
const useFocusMode = shouldUseFocusMode();
if (useFocusMode) {
  myMinimap.enableFocusMode();
} else {
  myMinimap.disableFocusMode();
}
```

vs
```js
myMinimap.setFocusModEnabled(shouldUseFocusMode());
```

But I could see an argument for this being clearer as well! Give it a think and pick whichever one you prefer =)

Reply:
Totally see the value of both the options. However, focus mode is on by default so there aren't many use cases for `enable.FocusMode()`. The developer would end up doing:
```
if (!useFocusMode) {
  myMinimap.disableFocusMode();
}
```

On another note, when it comes to API design, is there a reason not to provide both functionalities? That way they could use any of the two methods.

And, would it be good to also add a method that returns if it's enabled not, for example `myMinimap.isEnabled()`?

Reply:
> On another note, when it comes to API design, is there a reason not to provide both functionalities? That way they could use any of the two methods.

Adding multiple ways of doing things has a few draw backs:
  1) More maintainance for us (although in this case it would be small)
  2) More overhead for external developers (they have to decide what's right for their use case)
  3) More overhead for external developers (they have to read it both ways)

This is deifnitely something that people fall on either side of though! E.g. Perl (which Christopher can rave at you about :P ) is very much based around giving programmers the flexibility to do things in whichever way they want to. While Python does the opposite "There should be one-- and preferably only one --obvious way to do it."

I personally fall on the "less is better" side of things!

> And, would it be good to also add a method that returns if it's enabled not, for example myMinimap.isEnabled()?

I think that is a wonderful idea =)